File: very-easy/desafio-05.js
// Escreva uma função que recebe uma string e retorna a maior letra segundo a ordem alfabética em minúsculo. Assuma que a string não possui nenhuma letra com acento, número ou caractere especial, apenas letras e espaços.
const stringSearch = word => {
    const alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"];
    let wordNormalize = word.normalize('NFD').replace(/[\u0300-\u036f]/g, '').toLowerCase();
    let wordSearchLetter = wordNormalize.split("")
    let highLetter = 0;
    for (let indexWord = 0; indexWord < word.length; indexWord++) {
        for (let indexAlphabet = 0; indexAlphabet < alphabet.length; indexAlphabet++) {
            if (wordSearchLetter[indexWord].indexOf(alphabet[indexAlphabet]) != -1)
                if (indexAlphabet > highLetter)
                    highLetter = indexAlphabet;
        }

    }
    return alphabet[highLetter]
}

console.log(stringSearch("Ismael Soares Dória"));

File: very-easy/desafio_05.py
import unicodedata
# def search_letter_high(word):
#     alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l",
#                 "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"]


def no_space(word):
    """Escreva uma função que recebe uma string e retorna a maior letra segundo a ordem alfabética em minúsculo. Assuma que a string não possui nenhuma letra com acento, número ou caractere especial, apenas letras e espaços."""
    word_no_space_lower = unicodedata.normalize("NFKD", word).encode(
        "ASCII", "ignore").decode("ASCII").replace(" ", "").lower()
    alphabet = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l",
                "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"]
    high_letter = 0
    for index_word in range(len(word_no_space_lower)):
        for index_letter in range(len(alphabet)):
            if (word_no_space_lower[index_word].find(alphabet[index_letter]) != -1):
                if index_letter > high_letter:
                    high_letter = index_letter
    return alphabet[high_letter]


print(no_space("Ismáêl Soãres Dória"))
